File: .skills/openclaw-skills/skills/manuelhettich/ticktick/scripts/commands/lists.ts
import { api, type Project } from "../api";

interface ListsOptions {
  json?: boolean;
}

function formatProject(project: Project): string {
  const color = project.color ? ` (${project.color})` : "";
  const closed = project.closed ? " [closed]" : "";
  return `• ${project.name}${color}${closed}\n  id: ${project.id}`;
}

export async function listsCommand(options: ListsOptions): Promise<void> {
  try {
    const projects = await api.listProjects();

    if (options.json) {
      console.log(JSON.stringify(projects, null, 2));
      return;
    }

    if (projects.length === 0) {
      console.log("No projects found.");
      return;
    }

    console.log(`\nProjects (${projects.length}):\n`);
    for (const project of projects) {
      console.log(formatProject(project));
      console.log();
    }
  } catch (error) {
    console.error(
      `Error: ${error instanceof Error ? error.message : String(error)}`
    );
    process.exit(1);
  }
}

File: .skills/openclaw-skills/skills/hfarazi/ticktickpower/ticktick/commands/lists.py
import json
import sys

from ..api import api


def _format_project(project: dict) -> str:
    color = f" ({project['color']})" if project.get("color") else ""
    closed = " [closed]" if project.get("closed") else ""
    return f"• {project['name']}{color}{closed}\n  id: {project['id']}"


def lists_command(args) -> None:
    try:
        projects = api.list_projects()

        if args.json:
            print(json.dumps(projects, indent=2))
            return

        if not projects:
            print("No projects found.")
            return

        print(f"\nProjects ({len(projects)}):\n")
        for project in projects:
            print(_format_project(project))
            print()

    except RuntimeError as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)
